src/REveMu2eMainWindow.cc

```diff
@@ -252,84 +150,259 @@ void REveMu2eMainWindow::GeomDrawer(TGeoNode* node, REX::REveTrans& trans, REX::
-    }
+}
+
+void REveMu2eMainWindow::getOffsets(TGeoNode* n,const std::string& str, REX::REveTrans& trans, int maxlevel, int level, std::vector<std::pair<std::string, std::vector<float>>> & offsets) {
+  ++level;
+  if (level > maxlevel){
+    return;
+  }
+  std::string name(n->GetName());
+  j++;
 
-    if(geomOpt.showCaloCrystals){
-      static std::vector <std::string> substrings_crystals  {"caloCrystal"};
-      for(auto& i: substrings_crystals){
+  REX::REveTrans ctrans;
+  ctrans.SetFrom(trans.Array());
+  TGeoMatrix     *gm = n->GetMatrix();
+  const Double_t *rm = gm->GetRotationMatrix();
+  const Double_t *tv = gm->GetTranslation();
+  REX::REveTrans t;// NOTE: starts index at "1" whereas the matrices are indexed from 0
+  t(1,1) = rm[0]; t(1,2) = rm[1]; t(1,3) = rm[2];
+  t(2,1) = rm[3]; t(2,2) = rm[4]; t(2,3) = rm[5];
+  t(3,1) = rm[6]; t(3,2) = rm[7]; t(3,3) = rm[8];
+  t(1,4) = tv[0]  ; t(2,4) = tv[1]  ; t(3,4) = tv[2] ;
+
+  ctrans *= t;
+  std::cout<<j<<" "<<name<<" located at "<<tv[0]<<" "<< tv[1]<<" "<<tv[2]<<std::endl;
+  std::vector<float> pos;
+  std::pair<std::string, std::vector<float>> offset;
+  pos.push_back(tv[0]);
+  pos.push_back(tv[1]);
+  pos.push_back(tv[2]);
+  offset = make_pair(name, pos);
+  offsets.push_back(offset);
+  n->ls();
+  int ndau = n->GetNdaughters();
+  for ( int i=0; i<ndau; ++i ){
+    TGeoNode * pn = n->GetDaughter(i);
+      REX::REveTrans ctrans;
+      ctrans.SetFrom(trans.Array());
+      {
+        TGeoMatrix     *gm = n->GetMatrix();
+        const Double_t *rm = gm->GetRotationMatrix();
+        const Double_t *tv = gm->GetTranslation();
+        REX::REveTrans t;// NOTE: starts index at "1" whereas the matrices are indexed from 0
+        t(1,1) = rm[0]; t(1,2) = rm[1]; t(1,3) = rm[2];
+        t(2,1) = rm[3]; t(2,2) = rm[4]; t(2,3) = rm[5];
+        t(3,1) = rm[6]; t(3,2) = rm[7]; t(3,3) = rm[8];
+        t(1,4) = tv[0]  ; t(2,4) = tv[1]  ; t(3,4) = tv[2] ;
+       
+        ctrans *= t;
+        //std::cout<<j<<" "<<name<<" located at "<<tv[0]<<" "<< tv[1]<<" "<<tv[2]<<std::endl;
+        std::vector<float> pos;
+        std::pair<std::string, std::vector<float>> offset;
+        pos.push_back(tv[0]);
+        pos.push_back(tv[1]);
+        pos.push_back(tv[2]);
+        offset = make_pair(name, pos);
+        offsets.push_back(offset);
+        n->ls();
+      
+      }
+    getOffsets( pn, str, trans,  maxlevel, level, offsets);
+  }
+}
+
+void REveMu2eMainWindow::GeomDrawerSol(TGeoNode* node, REX::REveTrans& trans, REX::REveElement* beamlineholder, int maxlevel, int level, GeomOptions geomOpt, std::vector<std::pair<std::string, std::vector<float>>>& offsets) {
+
+  double x_sol = 0; double y_sol = 0; double z_sol = 0; 
+  double x_pt = 0; double y_pt = 0; double z_pt =0;
+  for(unsigned int i = 0; i < offsets.size(); i++){
+      if(offsets[i].first.find("DS3Vacuum0") != string::npos)      {
```

Comment:
The "0" is not part of the volume name.  It is part of the uniqueness decoration added by G4's GDML writer.  Please remove it.